# 1. Perkenalan

### Introduction

```
===================================================================================================================================
MILESTONE PHASE 0

Nama  : DZAKI AHMAD FARDIAN
Batch : 024

Pada Milestone ini saya akan melakukan analisis terhadap data yang diambil website penyedia dataset: Kaggle.com. Data yang saya gunakan adalah data sales perusahaan untuk barang Electronics & Accessories. Di sini saya akan melakukan penjabaran masalah, penyajian & pembersihan data, analisis data, visualisasi data dan pengambilan keputusan hasil analisis tersebut. 
===================================================================================================================================
```

Link to Tableau : https://public.tableau.com/app/profile/dzaki.af/viz/Milestone1_17345144833180/StatisticAnalysis?publish=yes

# 2. Identifikasi Masalah

## 1. Latar Belakang
    Dalam era digital yang berkembang pesat, e-commerce telah menjadi salah satu pilar utama dalam dunia bisnis modern. Perkembangan teknologi informasi, terutama internet, telah mengubah cara masyarakat berbelanja dan bertransaksi. Transformasi ini tidak hanya memberikan dampak signifikan terhadap pola konsumsi, tetapi juga membuka peluang baru bagi pelaku usaha, baik skala kecil maupun besar.
    Banyaknya kompetitor yang memanfaatkan teknologi sebagai bantuan dalam berjualan membuat perusahaan ingin bersaing dalam e-commerce.
    Perusahaan berencana memaksimalkan penjualan. Agar dapat bersaing dengan penjual lain, perusahaan ingin memproduksi barang yang sedang populer di pasar. Maka pihak perusahaan membutuhkan data hasil analisis kinerja produk.

## 2. S.M.A.R.T:
1. Spesific   : Perusahaan akan meiningkatkan penujualan dengan mencari tau product apa yang paling banyak terjual.
2. Measurable : Perusahaan ingin meningkatkan penjualan barang sebesar 20%.
3. Achievable : Perusahaan menyesuaikan barang dagangan dengan barang yang sedang populer di pasar.
4. Relevant   : Meningkatkan pendapatan jika perusahaan bisa mengetahui barang terpopuler dari sebuah sample.
5. Time-Bound : Perubahan ini dapat di capai selama 3 bulan. Bisa kita jabarkan menjadi : 
    1. Minggu 1-2 penyusunan strategi analisis.
    2. Minggu 3-4 pengumpulan data.
    3. Minggu 5-6 Analisis data.
    4. Minggu 7-8 Perencanaan strategi baru untuk bisnis 
    5. Minggu 9-11 Implementasi Strategi baru.
    6. Minggu 12 Evaluasi hasil pengimplementasian strategi baru.

6. Problem Statement :
	Perushaan ingin meningkatkan penjualan sebesar 20%. Hal ini dilakukan dengan bantuan analisis korelasi antara Product Name dan Category Product dengan Quantity Terjual dan Harga Product. Dari data hasil analisis ini dapat dilihat barang apa yang populer di pasar. Peningkatan penjualan diperkirakan dapat dicapai dalam waktu 3 bulan.

7. User : Hasil analisis ini dapat digunakan oleh team pengembangan bisnis dari perusahaan agar dapat membuat strategi bisnis berdasarkan kesimpulan yang diambil dari hasil analisis.


## 3. Penjabaran Masalah
1. Statistik Inferensial:
    Di sini saya akan melakukan Uji Hipotesis untuk sebuah pertanyaan :
    - Apakah gender customer laki-laki (male) lebih banyak membeli product dari perempuan (female)?

2. Visualisasi Data:
    Akan ada beberapa visualisasi data seputar pertanyaan berikut ini :
    - Product apa yang terjual paling banyak?
    - Category apa yang terjual paling banyak ?
    - Category apa yang sering di Return customer?
    - Bagaimana tren penjualan dari waktu ke waktu?

3. Statistik Deskriptif:
    Untuk statistik Deskriptif saya akan melakukan sebuah Uji Korelasi untuk menjawab pertanyaan berikut :
    - Apakah customer cenderung memilih product dengan harga murah?

# 3. Data Loading

Data set yang saya gunakan berasal dari Kaggle.com berikut link datasetnya : 

https://www.kaggle.com/datasets/brsahan/e-commerce-dataset/data

Import Libary yang akan digunakan

In [1]:
import pandas as pd
from scipy import stats
import plotly.express as px

Melakukan read data.

In [2]:
ms = pd.read_csv('realistic_e_commerce_sales_data.csv')

Menampilkan data.

In [3]:
ms

Customer ID  Gender Region   Age Product Name     Category  Unit Price  \
0      CUST0268    Male  North   NaN      Monitor  Electronics       300.0   
1      CUST0046    Male   West  22.0   Headphones  Accessories       100.0   
2      CUST0169  Female  South  54.0      Monitor  Electronics       300.0   
3      CUST0002    Male  North  23.0   Headphones  Accessories       100.0   
4      CUST0173  Female  South   NaN       Laptop  Electronics      1500.0   
..          ...     ...    ...   ...          ...          ...         ...   
995    CUST0201  Female  South  49.0   Headphones  Accessories       100.0   
996    CUST0133    Male   East  47.0       Laptop  Electronics      1500.0   
997    CUST0055  Female  North   NaN        Mouse  Accessories        30.0   
998    CUST0023  Female  South  29.0       Laptop  Electronics      1500.0   
999    CUST0040    Male    NaN  29.0   Smartwatch    Wearables       200.0   

     Quantity  Total Price  Shipping Fee Shipping Status  Order Date  
0           5         1500         13.31        Returned  2023-12-08  
1           2          200          6.93      In Transit  2023-04-09  
2           1          300         11.31        Returned  2023-08-28  
3           5          500         12.22       Delivered  2023-01-18  
4           3         4500          5.40       Delivered  2023-01-19  
..        ...          ...           ...             ...         ...  
995         1          100         17.21      In Transit  2023-01-05  
996         1         1500         19.19       Delivered  2023-04-01  
997         5          150         19.35       Delivered  2023-10-20  
998         5         7500         10.36        Returned  2023-01-07  
999         1          200          7.53       Delivered  2023-01-30  

[1000 rows x 12 columns]

## Penjabaran data secara general

Data ini adalah data penjualan barang electronic dan gadget. Di sini kita dapatkan column :
- Customer ID : Sebagai Primary Key dari tiap transaksi.
- Gender : Kelamin customer.
- Region : Wilayah tempat tinggal customer.
- Age : Umur customer.
- Product Name : Untuk Product Name isinya bukan langsung nama productnya (ex: Laptop Lenovo...) melainkan jenis barangnya. Seperti Laptop, Monitor, Headphones, Smartphone.
- Category : Toko memiliki category sendiri yaitu Electronics (untuk Monitor, Laptop), Accessories (untuk Headphones) dan Wearables (Smartwatch).
- Unit Price : Harga per-unit.
- Quantity : Jumlah barang yang dibeli customer.
- Total Price : Jumlah yang harus dibayarkan customer untuk semua barang per-transaksi.
- Shipping Fee : Harga jasa pengiriman.
- Shipping Status : Status pengiriman barang.
- Order Date : Tanggal dilakukannya order.

## Missing Value dan Data Type.

Saya akan menggunakan function .info() untuk menampilkan informasi dari dataframe.

In [4]:
ms.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Customer ID      1000 non-null   object 
 1   Gender           1000 non-null   object 
 2   Region           950 non-null    object 
 3   Age              900 non-null    float64
 4   Product Name     1000 non-null   object 
 5   Category         1000 non-null   object 
 6   Unit Price       1000 non-null   float64
 7   Quantity         1000 non-null   int64  
 8   Total Price      1000 non-null   int64  
 9   Shipping Fee     1000 non-null   float64
 10  Shipping Status  950 non-null    object 
 11  Order Date       1000 non-null   object 
dtypes: float64(3), int64(2), object(7)
memory usage: 93.9+ KB


Berikut untuk memastikan berapa jumlah pasti nilai null.

In [5]:
ms.isnull().sum()

Customer ID          0
Gender               0
Region              50
Age                100
Product Name         0
Category             0
Unit Price           0
Quantity             0
Total Price          0
Shipping Fee         0
Shipping Status     50
Order Date           0
dtype: int64

**KESIMPULAN :**

Tidak ada nilai duplicated.
Di sini kita bisa lihat terdapat data yang bernilai 950, 950 dan 900. Artinya ada data yang salah satu kolomnya berisikan null. Nanti saya akan lakukan penghapusan pada row tersebut di tahap data cleaning. Kemudian untuk 'Order Date' bertipe data object. Saya akan ubah menjadi Date.

## Duplicated Value

Berikut saya menggunakan function duplicated untuk mengecek duplicated dan saya aggregasi sum untuk nilai duplicatednya. 

In [6]:
ms.duplicated().sum()

np.int64(0)

**Kesimpulan :**

Tidak ada nilai duplicated.

# 4. Data Cleaning

## Handling Missing Value

Sebelumnya pada column 'Region', 'Age' dan 'Shipping Status' aku gunakan function dropna() untuk menghapus row yang terdapat nilai nullnya.

In [7]:
ms.dropna(subset=['Region'], inplace=True)
ms.dropna(subset=['Age'], inplace=True)
ms.dropna(subset=['Shipping Status'], inplace=True)

## Change Data Type

Di bawah ini saya ubah tipe data 'Order Date' menjadi Date.

In [8]:
ms['Order Date'] = pd.to_datetime(ms['Order Date'])

Di sini kita bisa lihat nilai rownya menjadi 810. Artinya terdapat 190 row yang terhapus. Dan untuk 'Order Date' sudah menjadi datetime.

In [9]:
ms.info()

<class 'pandas.core.frame.DataFrame'>
Index: 810 entries, 1 to 998
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Customer ID      810 non-null    object        
 1   Gender           810 non-null    object        
 2   Region           810 non-null    object        
 3   Age              810 non-null    float64       
 4   Product Name     810 non-null    object        
 5   Category         810 non-null    object        
 6   Unit Price       810 non-null    float64       
 7   Quantity         810 non-null    int64         
 8   Total Price      810 non-null    int64         
 9   Shipping Fee     810 non-null    float64       
 10  Shipping Status  810 non-null    object        
 11  Order Date       810 non-null    datetime64[ns]
dtypes: datetime64[ns](1), float64(3), int64(2), object(6)
memory usage: 82.3+ KB


Saya lakukan pengecekkan kembali untuk nilai nullnya.

In [10]:
ms.isnull().sum()

Customer ID        0
Gender             0
Region             0
Age                0
Product Name       0
Category           0
Unit Price         0
Quantity           0
Total Price        0
Shipping Fee       0
Shipping Status    0
Order Date         0
dtype: int64

## Export cleaned data.

Data ini saya export menjadi format xlsx yang nanti akan digunakan pada pembuatan dashboard.

In [11]:
ms.to_excel("cleaned_sales_data.xlsx")

# 5. Analisis dan Perhitungan

## Skewness

Nilai Skewness akan menentukan kemiringan sebuah data. Skewness dari skewness kita bisa mengkategorikan sebuah data sebagai 
- Approximately Symmetric (-0,5 sampai 0,5)
- Moderately Skewed (-1 sampai -0,5 dan 05 sampai 1)
- Highly skewed (kurang dari -1 dan lebih dari 1). 

Plus-Minus di sini menandakan data akan condong ke arah kanan atau kiri (Positive atau Negative)

In [12]:
# Saya membuat variable skew untuk masing-masing column dengan nilai skew yang diambil dari function .skew().
skew_unit_price = ms['Unit Price'].skew()
skew_quantity = ms['Quantity'].skew()
skew_total_price = ms['Total Price'].skew()

# tampilkan variablenya.
print(f' Skew  Unit Price : {skew_unit_price}')
print(f' Skew Quantity : {skew_quantity}')
print(f' Skew Total Price : {skew_total_price}')

 Skew  Unit Price : 1.5044121674676028
 Skew Quantity : -0.018366427444406756
 Skew Total Price : 2.0085177038694453


**KESIMPULAN:** 

Di atas ini untuk nilai skewnya. Untuk Unit Price dan Total Price, memiliki nilai lebih dari -0.5 atau lebih dari 0.5, jadi datanya sudah dapat dipastikan tidak symmetric. Pada skew ini kita bisa simpulkan bahwa datanya miring ke kanan (Positively Skew) dan Highly Skew. Namun untuk Quantity, data dikategorikan sebagai Approximately Symetric. Dari sini kita bisa mengira-ngira bahwa nilai data tidak Normal.

## Kurtosis

Nilai kurtosis juga digunakan untuk melihat apakah data terdistribusi dengan normal. Pada kurtosis ini kita bisa mendapat kesimpulan apakah datanya memiliki Outlier. Outlier ini artinya ada data yang berada jauh dari kumpulan data lainnya. Kalau banyak data yang menjadi outlier, maka data ini tidak terdistribusi dengan normal. Ada tiga jenis hasil kurtosis.
- Kurotis Leptokurtic, nilainya lebih dari 3. Leptokurtic ini kalau dalam penggambarannya, memiliki puncak yang tajam. Pada kurtosis ini lebih banyak memiliki outlier dari kurtosis lainnya.
- Kurtosis Mesokurtic, nilainya sama dengan 3. Untuk Mesokurtic gambarannya mirip distribusi normal. Jumlah outliernya tidak signifikan.
- Kurtosis Platikurtic, nilanya kurang dari 3. Untuk Platikurtic puncak pada gambarnya datar. Data lebih terkonsentrasi sehingga variasinya kecil.

In [13]:
# Saya membuat variable skew untuk masing-masing column dengan nilai skew yang diambil dari function .kurtosis()
kurtosis_unit_price = ms['Unit Price'].kurtosis()
kurtosis_quantity = ms['Quantity'].kurtosis()
kurtosis_total_price = ms['Total Price'].kurtosis()

# tampilkan variablenya
print(f' Kurtosis  Unit Price : {kurtosis_unit_price}')
print(f' Kurtosis Quantity : {kurtosis_quantity}')
print(f' Kurtosis Total Price : {kurtosis_total_price}')

 Kurtosis  Unit Price : 1.8478917038097245
 Kurtosis Quantity : -1.2894363537964701
 Kurtosis Total Price : 3.473910848456027


**KESIMPULAN:**

Untuk Unit Price dan Total Price di sini memiliki nilai Kurtosis yang berbeda. Di sini Unit Price dikategorikan sebagai Platikurtic sedangkan Total Price sebagai Leptokurtic. Artinya Unit Price nilainya lebih terkonsentrasi, hal ini disebabkan karena data berasal dari 1 perusahaan dan variasi harganya yang sedikit.
 
Sedangkan untuk Total Price, karena customer membeli product yang berbeda-beda dengan jumlah yang berbeda-beda juga. Maka dari itu Total Price memiliki banyak Outlier. Yang terakhir Quantity dikategorikan sebagai Platikurtic. 

Di sini Quantity lebih terkonsentrasi dari pada Unit Price. Hal ini saya rasa dikarenakan untuk Quantity yang dibeli tidak akan terlalu banyak, mengingat product yang dijual adalah electronic dan wearables. Tidak mungkin seorang customer membeli SmartPhone sebanyak 1000 dalam sekali transaksi, kecuali distributor. Tetapi Distributor tidak beli dari toko langsung. Jadi kemungkinan ada variasi sangat kecil.

## Uji Normalitas untuk menentukan pemakaian metode saat Uji Hipotesis dan Uji Korelasi.

Sebelumnya saya sudah menampilkan Skewness dan Kurtosis yang menandakan terdapat variasi yang sangat banyak untuk data-datanya. Jadi sebelum saya melakukan Uji Hipoteses saya akan menguji Normalitas Data. Uji Normalitas ini dilakukan agar saya bisa menentukan metode yang nantinya saya akan pakai untuk Uji Hipotesis dan Uji Korelasi. Karena beberapa metode hanya bisa digunakan pada saat Data terdistribusi tidak normal. Di bawah ini saya menggunakan metode Shapiro untuk Uji Normalitas.

### Uji Normalitas Quantity.

Quantity akan dipakai pada Uji Hipotesis dan juga Uji Korelasi.

In [14]:
# buat variable yang isinya menggunakan function .shapiro() dari scipy untuk uji normalitas pada column Quantity.
res = stats.shapiro(ms['Quantity'])
print(f'p-value: {res.pvalue}')

p-value: 9.074856616240531e-24


**KESIMPULAN :**

Dari hasil Uji Normalitas di atas, saya dapatkan nilai : 9.074856616240531e-24. Nilai ini lebih kecil dari signifikan level: 0,05. Dari sini kita bisa mengetahui bahwa data Quantity ini tidak terdistribusi dengan normal. Sehingga untuk Uji Hipotesis saya akan menggunakan Uji Mann-Whitney U. 

### Uji Normalitas Unit Price.

Di bawah ini saya uji juga untuk nilai Unit Price yang nanti akan dipakai saat Uji Korelasi.

In [15]:
# buat variable yang isinya menggunakan function .shapiro() dari scipy untuk uji normalitas pada column Unit Price.
res = stats.shapiro(ms['Unit Price'])
print(f'p-value: {res.pvalue}')

p-value: 4.820219008266184e-34


**KESIMPULAN :**

Hasilnya Unit Price juga tidak terdistribusi dengan normal. Untuk Uji Korelasi akan saya gunakan metode Spearman. Karena Spearman dipakai saat data tidak terdistribusi dengan normal dan data numerical.

## Uji Hipotesis - Statistik Inferensial 

Di sini saya akan menganalisis sebuah pertanyaan yang muncul menggunakan Statistik Inferensial.

'Apakah gender customer laki-laki (male) lebih banyak membeli product dari perempuan (female)?'

 Sebelumnya saya menyebutkan akan menggunakan Uji Mann-Whiteney U. Alasan penggunaan uji ini ini selain data tidak terdistribusi tidak normal, saya akan menggunakan 2 variable yang bersifat Categorical (Gender) dengan Numerical (Quantity).

Di sini saya menggunakan Hipotesis berikut ini.

### Hipotesis

- H0 : TIDAK ADA perbedaan Quantity terjual dari customer male dengan customer female.
- H1 : ADA perbedaan Quantity terjual dari customer male dengan customer female.

Jika nilai P-Value yang saya dapat kurang dari 0.05 maka H0 akan ditolak dan kesimpulan yang diambil adalah H1. Sebaliknya Jika P-Value lebih dari 0.05, maka H0 akan diterima sebagai kesimpulan. 

In [16]:
# Saya import function mannwhiteyu untuk melakukan uji hipotesis.
from scipy.stats import mannwhitneyu

# Saya buat variable untuk male dan female terpisah, yang isinya adalah Quantity yang sesuai dengan nilai Column Gender (male/female)
male_quantity = ms[ms["Gender"] == "Male"]["Quantity"]
female_quantity = ms[ms["Gender"] == "Female"]["Quantity"]

# gunakan function mannwhiteyu dengan argument yang berisikan 2 variable tadi.
# Di sini alternative='to-sided' untuk mengevaluasi median dari kedua kelompok.
stat, p = mannwhitneyu(male_quantity, female_quantity, alternative='two-sided')
print(f"U-Statistic: {stat}, P-Value: {p}")

U-Statistic: 85170.0, P-Value: 0.2966645199833884


### Kesimpulan
**KESIMPULAN :**

Karena nilai P-Value lebih dari 0.05, maka H0 diterima.

TIDAK ADA Perbedaan Quantity terjual customer male dengan customer female.

## Visualisasi

Berikut ini pertanyaan yang tadi saya jabarkan.

- Question 1 : Product apa yang terjual paling banyak?
- Question 2 : Category apa yang terjual paling banyak ?
- Question 3 : Category apa yang sering di Return customer?
- Question 4 : Bagaimana tren penjualan dari waktu ke waktu?

Saya ingin melihat penggambaran untuk masing-masing pertanyaan. Penggambaan ini akan memudahkan team pengembangan bisnis untuk membuat rencana pengembangan bisnis ini.

### Visualisasi1 (Bar Chart : Menampilkan product dengan jumlah penjualan tertinggi)

In [17]:
# Di sini saya membuat variable top_product dan akan melakukan groupby untuk product name dengan quantity 
# yang di aggregasi dengan sum.
top_products = ms.groupby("Product Name")["Quantity"].sum().reset_index()
# Dari variable itu kemudian saya mensort (sort_values) nilai quantity (by="Quantity") tersebut dari atas ke bawah (ascending=Flase)
# dan ditampilkan 5 data teratas (head(5))
top_products = top_products.sort_values(by="Quantity", ascending=False).head(5)

# saya buat variable fig yang nilainya menggunakan function .bar() dari class plotly.
# isi argumentnya adalah attribute dari chart.
# top_products, nilai variable yang saya buat tadi. Karena saya mau memunculkan datanya ke chart jadi saya panggil.
# x dan y adalah sumbu chartnya.
# title untuk judul di atas chart.
# lables untuk menamai nilai pada chart. Jadi saat cursor dihover ke arah bar dari barchart, akan muncul nama dan nilainya.
# text untuk memberi text di atas chartnya.
fig = px.bar(top_products, x="Product Name", y="Quantity", 
             title="Top 5 Products by Quantity Sold", 
             labels={"Product Name": "Product Name", "Quantity": "Total Quantity"},
             text="Quantity")
# function update_layout untuk mengubah ukuran dan posisi. Di sini saya mengubah posisi dari title yang sudah dibuat.
# dan saya lakukan pengubahan pada tinggi (height) karena tanpa pengubahan ini Quantity 423 akan terpotong.
fig.update_layout(title_x = 0.5, autosize=False, height=600)
# di bawah ini saya melakukan penempatan untuk text yang di dalam argument .bar().
# di sini saya tempatkan di luar chart (outside)
fig.update_traces(textposition="outside")
fig.show()

#### Kesimpulan dan Insight
**KESIMPULAN :**

Di atas ini merupakan visualisasi hubungan data Product Name dengan Quantity. Di sini saya tampilkan 5 product teratas. Di dapatkan secara berurut Monitor, Keyboard, Laptop, Smartphone dan Mouse.daripada Laptop.

Kita bisa lihat lebih banyak yang membeli Monitor. Hal ini saya rasa karena beberapa pengguna Laptop membeli Monitor tambahan untuk pekerjaan. Tetapi tidak semua pengguna Laptop membutuhkan Monitor tambahan, namun Monitor dibutuhkan pengguna PC. Sehingga jumlah pembeli Monitor lebih banyak. 
 
Begitu juga dengan Keyboard dan Laptop. Pembeli Keyboard belum tentu untuk PC, bisa untuk Laptop. Sedangkan pengguna Laptop belum tentu membeli Keyboard. Jadi pembelian Keyboard lebih besar.

**INSIGHT :**

Dari data ini bisa menjadi referensi untuk team pengembangan bisnis dalam memilih product apa yang perlu dijual.

### Visualisasi2 (Pie Chart : Distribusi penjualan berdasarkan Category)

In [18]:
# Penjelasan untuk semua code visualisi akan mirip. Jadi saya persingkat.
# Grouby dan sum.
category_sales = ms.groupby("Category")["Quantity"].sum().reset_index()

# Masukkan attribute chartnya. Di sini pakai Pie Chart.
fig = px.pie(category_sales, names="Category", values="Quantity", 
             title="Sales Distribution by Category", 
             labels={"Category": "Product Category", "Quantity": "Total Quantity"})
# Posisikan title dan tambah tingginya. Walaupun tidak ada nilai yang terpotong, tinggi ditambah agar bentuk semua Chart sama.
fig.update_layout(title_x = 0.5, autosize=False, height=600)
fig.show()

#### Kesimpulan dan Insight
**KESIMPULAN :**

Di atas ini Pie Chart untuk jumlah sales yang dibagi berdasarkan category. Kita bisa lihat masing-masing urutannya dari Electronics (45.9%), Accessories (41%) dan Wearables(13.1%).

Untuk kesimpulan chart ini sebenarnya kita bisa kira-kira dari Chart sebelumnya. Top 5 Product yang terjual Monitor (Electronics), Keyboard (Accessories), Laptop (Electronics), SmartPhone (Electronics), Mouse (Accessories). Yang tidak ada pada top 5 adalah Smartwatch (Wearables) dan Headphones (Accesories). Dari sini kita bisa mengetahui top 5 product di dominasi Electronics dan Accessories. 

Tapi untuk lebih jelas saya buat chart ini agar memisahkan dari Top 5 Product Chart. Karena saya rasa kalau digabungkan akan terlalu complex dan membuat pembaca chart kebingungan.

**INSIGHT :**

Walaupun Category Wearables terlihat sedikit. Hal ini disebabkan oleh sedikitnya product yang dikategorikan sebagai Wearables. Jika kita bandingkan terdapat 3 Electronics, 3 Accessories dan 1 Wearables. Seperti yang saya sebut untuk insight di 'Visualisasi2' bahwa jumlah sales tiap product tidak terlalu berbeda, yaitu sekitar 304 sampai 423. Untuk Smartwatch sendiri yang satu-satunya Wearables memiliki total Quantity terjual 318. Jadi saya rasa nilai yang di dapat untuk Wearables (13.1%) tidak menandakan sales Wearables sedikit. Mungkin untuk perusahaan **dapat menambahkan 2 product yang bisa dikategorikan sebagai Wearables agar bisa dibandingkan dengan Category lain**.

### Visualisasi3 (Bar Chart : Category product yang Sering direturn)

In [19]:
# dapatkan data yang Shipping Satus = Returned.
returned_data = ms[ms["Shipping Status"] == "Returned"]
# groupby Category dan sort berdasarkan count return.
returned_categories = returned_data.groupby("Category").size().reset_index(name="Return Count").sort_values(by="Return Count", ascending=False)

# masukkan attribute chart. saya pakai bar chart.
fig = px.bar(returned_categories, x="Category", y="Return Count", 
             title="Top Returned Categories", 
             labels={"Category": "Product Category", "Return Count": "Number of Returns"},
             text="Return Count")

# posisi title dan tinggi.
fig.update_layout(title_x = 0.5, autosize=False, height=600)
# penempatan text di luar.
fig.update_traces(textposition="outside")
fig.show()

#### Kesimpulan dan Insight
**KESIMPULAN :**

Di atas ini adalah jumlah transaksi yang dikembalikan untuk Category tertentu. Didapatkan Electronics returned sebanyak 122, Accessories sebanyak 103, dan Wearables sebaganyak 33. Pada Kesimpulan & Insight Visualisasi 1 dan 2 saya sudah menyebutkan bahwa Wearables hanya terdapat 1 product dibanding Electronics yang terdapat 3 product. Karena perbandingan jumlah product pada kategori ini 1:3 dan jumlah product returned juga 1:3 untuk Electronics dan Wearables. **Maka kita bisa simpulkan bahwa jumlah transaksi pada tiap product hampir sama**

**INSIGHT :**

Masih sama seperti tadi, perusahaan dapat menambah product pada Category agar kita bisa melihat lebih jelas Category mana yang tidak populer.

### Visualisasi4 (Line Chart : Trend penjualan berdasarkan waktu. )

In [ ]:
# ubah Order Date menjadi datetime, untuk memastikan.
ms["Order Date"] = pd.to_datetime(ms["Order Date"])
# groupby order date dengan quantity
# order date diambil month dan yearnya (["Order Date"].dt.to_period("M"))
# quantity di sum()
# reset_index agar urut tanggalnya
sales_trend = ms.groupby(ms["Order Date"].dt.to_period("M"))["Quantity"].sum().reset_index()
# order dalam variable tadi diubah menjadi string agar memastikan bisa diubah
sales_trend["Order Date"] = sales_trend["Order Date"].astype(str)

# masukkan attribute chart. saya pakai line chart
fig = px.line(sales_trend, x="Order Date", y="Quantity", 
              title="Monthly Sales Trend", 
              labels={"Order Date": "Order Month", "Quantity": "Total Quantity"})
# update posisi title dan tinggi.
fig.update_layout(title_x = 0.5, autosize=False, height=600)
fig.show()

#### Kesimpulan dan Insight
**KESIMPULAN :**

Data yang transaksi yang ada pada Dataset hanya mencakup sales selama 1 tahun. Dari yang saya lihat ada sebuah tren naik-turun untuk sales setiap 3 bulan. Bisa kita lihat dari Januari **->Februari, Maret->** April sales mengalami **penurunan lalu pada bulan Mei sales naik**. Kemudian dari Mei **->Juni, Juli->** Agustus sales menurun kembali. Setelah itu **sales naik lagi pada bulan September**. Terakhir kembali turun dari October **->November->** Desember.

**INSIGHT :**

Team pengembangan bisnis perlu menghandle penurunan sales pada bulan-bulan tersebut. Saya memiliki dua saran. **Saran pertama saya, untuk memberi diskon atau promo untuk bulan tersebut. Kedua memberi sebuah bundle untuk product. Untuk saran pertama ini akan saya lakukan Uji Korelasi antara harga murah dengan jumlah quantity terjual**

## Uji Korelasi - Statistik Deskriptif

Sebelumnya terdapat insight di mana saya menyarankan pemberian diskon untuk product di bulan-bulan tertentu. Di sini saya akan mencari tau :
'Apakah customer lebih memilih product dengan harga murah?'

Uji korelasi yang akan saya gunakan di analisis ini adalah Uji Spearman. Spearman ini akan di uji untuk Unit Price dengan Quantity. Spearman ini digunakan karena Unit Price dan Quantity merupakan sebuah nilai numerical. Saya tidak menggunakan Pearson karena dari hasil Uji Normalitas sebelumnya disimpulkan bahwa data ini tidak terdistribusi dengan normal.

In [ ]:
# Di sini saya menggunakan function .spearmanr() dari class scipy. Dengan argument berisikan Unit Price dan Quantity.
# Nilainya dimasukkan ke variable corr_rho dan pval_s.
corr_rho, pval_s = stats.spearmanr(ms['Unit Price'], ms['Quantity'])
print(f"rho-correlation: {corr_rho:.2f}, p-value: {pval_s}")

rho-correlation: -0.04, p-value: 0.23532441239581475


### Kesimpulan dan Insight

Kita dapatkan nilai rho sebesar -0.04. Nilai ini mendekati 0 yang menunjukkan hubungan yang sangat lemah.

Untuk nilai P-Value di dapatkan 0.235. Karena P-Value > 0.05 maka

**KESIMPULAN :**

Hubungan antara Unit Price dan Quantity tidak signifikan. Yang artinya harga dari sebuah barang tidak menjamin barang tersebut terjual dengan banyak. 

# 6. Pengambilan Kesimpulan

Dari hasil analisis yang saya lakukan sebelumnya. Bisa diambil sebuah kesimpulan dan insight dengan point-point berikut ini:


## - Statistik Inferensial

**KESIMPULAN dan INSIGHT**

Pada analisis Statistik Inferensial ditemukan bahwa tidak ada hubungan sama sekali antara gender dengan banyaknya product yang terjual. Oleh karena itu **team pengembang bisnis tidak perlu fokus untuk mengembangkan product yang disukai oleh gender tertentu**

## - Visualisasi 1 - Top 5 Products

**KESIMPULAN dan INSIGHT**

Dari Chart Top 5 Products saya dapatkan bahwa Monitor lebih banyak dibutuhkan oleh customer. Seperti yang saya sebutkan sebelumnya pada bagian 'visualisasi1', bahwa customer yang membeli Monitor bukan berarti pengguna PC, melainkan pengguna Laptop. Dengan banyaknya calon pembeli Monitor, Monitor menjadi barang yang akan lebih banyak dibeli. Sehingga **team pengembangan bisni dapat memberi promo untuk product lain. Misalnya Buy 2 Get 1 atau buy product A get product B, untuk product-product yang kurang populer.**

## - Visualisasi 2 - Sales Distribution by Category

**KESIMPULAN dan INSIGHT**

Seperti yang saya tuliskan pada analisis visualisasi 2. Bahwa Category ini belum bisa digunakan sebagai perbandingan karena product yang terdapat pada Category Wearables hanya terdapat Smartwatch. Dibandingkan Electronics yang memiliki 3 product dan Accessories memiliki 3 product. perbandingan untuk Electronics:Accessories:Wearables adalah 1116:996:318. Mungkin jika kita membandingkan Electronics dan Accessories akan adil karena jumlah perbandingan Product Name pada kedua Category tersebut adalah 1:1. Di sini kita bisa melihat bahwa Electronics memiliki peforma lebih dari Accessories. Namun jika kita membandingkan Electronics dengan Wearables, perbandingan Product Name nya adalah 3:1. Jadi kalau saya ingin membandingkan peforma product tersebut, saya harus membagi 3 penjualan Electronics agar perbandingannya setara. Hal ini saya lakukan karena pada data penjualan untuk Smartwatch (Wearables) berada pada 318 yang hampir sama dengan kebanyakan product dengan nilai terjual 300+ (kecuali Monitor, 400+). **Jadi jika kita anggap dengan membagi 3 penjualan category Electronics dan Accessories, akan di dapatkan 372:332:318. Category Wearables masih memiliki nilai jual lebih rendah daripada Category lainnya. Jika team pengembangan bisnis tidak masalah dengan asumsi ini, maka kesimpulannya Category Electronics yang memiliki peforma terbagus.** Akan tetapi menurut saya pribadi hal ini tidak begitu akurat dalam mengambil kesimpulan, jadi akan lebih bagus jika **perusahaan menambah product baru dengan Category Wearables sehingga lebih mudah jika ingin melihat perbandingannya atau menghapus Category Wearables dan menggambungkan Smartwatch ke Category Accessories/Electronics.** Jika team pengembangan bisnis tetap memakai asumsi sales/3. Saran saya masih sama seperti Visualisasi 1 **tambahkan bundles atau promo untuk category yang tidak populer.**

## - Visualisasi 3 - Most Returned Product by Category

**KESIMPULAN dan INSIGHT**

Sama seperti Visualisasi 2. Perbandingan untuk returned product by category adalah 122:103:33(Electronics:Accessories:Wearables). Jika asumsi dibagi 3 untuk Electronics dan Wearables akan menjadi 40:34:33. **Product Electronics paling sering dikembalikan oleh customer**. Oleh karena itu **team pengembangan bisnis harus mengevaluasi antara kualitas barang yang dikirim atau kualitas jasa pengiriman. Hal ini bisa dilakukan dengan pengambilan survey atau feedback dari customer yang ingin mengembalikan product.** 

## - Visualisasi 4 - Monthly Sales Trend

**KESIMPULAN dan INSIGHT**

Penjelasan Line Chart teridentifikasi penaikan dan penurunan barang dengan interval waktu yang berpola, yaitu setiap 3 bulan. **Untuk team pengembangan bisnis sebaiknya membuat sebuah promo atau diskon untuk bulan-bulan terjadinya penurunan.** Akan tetapi karena Hubungan antara harga murah dengan quantity terjual belum di uji, **maka bisa dipilih jalur promo. Promo di sini tidak selalu hanya diskon atau pun buy 1 get 1. Promo bisa dilakukan dengan menghubungkan season, event ataupun kolaborasi dengan product atau influencer.** Diharapkan agar sales naik untuk bulan-bulan tersebut.

## - Statistik - Descriptive

**KESIMPULAN dan INSIGHT**

Dari hasil analisis Statistik Descriptive saya mendapatkan bahwa harga sebuah product tidak mempengaruhi banyaknya product terjual. Maka dari itu **team pengembangan bisni tidak perlu meng-improve harga product. Team pengembangan bisnis bisa cari alternatif lain seperti pembuatan bundling.**